src/FRP/Event/Time.purs

```diff
@@ -15,3 +17,38 @@ foreign import animationFrame :: Event Unit
 
 -- | Create an event which reports the current time in milliseconds since the epoch.
 foreign import withTime :: forall a. Event a -> Event { value :: a, time :: Number }
+
+-- | Provided an input event and transformation, block the input event for the
+-- | duration of the specified period on each output.
+withBlocking
+  :: forall a b.
+     (Event a -> Event { period :: Number, value :: b })
+  -> Event a
+  -> Event b
+withBlocking process event
+  = fix \allowed ->
+      let
+        processed :: Event { period :: Number, value :: b }
+        processed = process allowed
+
+        expiries :: Event Number
+        expiries = pure 0.0 <|>
+          map (\{ time, value } -> time + value)
+              (withTime (map _.period processed))
+
+        comparison :: forall r. Number -> { time :: Number | r } -> Boolean
+        comparison a b = a < b.time
+
+        unblocked :: Event { time :: Number, value :: a }
+        unblocked = gateBy comparison expiries stamped
```

Comment:
Could you please inline `comparison` here? I think it would make it slightly easier to read.

Reply:
No worries!